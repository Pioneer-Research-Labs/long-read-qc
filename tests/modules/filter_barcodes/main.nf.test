nextflow_process {

    name "Test Process filter_barcodes"
    script "modules/filter_barcodes/main.nf"
    process "filter_barcodes"

    test("Should filter barcodes based on min/max lengths specified in config") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/Gateway_EmptyBarcodedLib1_20241217/barcodes_filtered.fasta").match()
        }

    }

}
