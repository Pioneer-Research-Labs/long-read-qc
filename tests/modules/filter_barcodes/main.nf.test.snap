{
    "Should filter barcodes based on min/max lengths specified in config": {
        "content": [
            "barcodes_filtered.fasta:md5,8c7dbffbb4d82dac42a1122464c59c62"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T12:02:13.385508"
    }
}